src/core/midiMsgFilter.cpp

```diff
@@ -0,0 +1,249 @@
+	if (n >= m_mask.size()) 
+		MidiMsgFilter::setFilterLength(n-1);
+
+	m_template.at(n) = b;
+}
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+void MidiMsgFilter::setMaskByte(unsigned int n, unsigned char b){
+	
+	// If filter is shorter than necessary,
+	// it is expanded with transparent bytes
+	if (n >= m_mask.size()) 
+		MidiMsgFilter::setFilterLength(n-1);
+	
+	m_mask.at(n) = b;
+}
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+void MidiMsgFilter::orTemplateByte(unsigned int n, unsigned char b,
+							unsigned int shl){
+
+	// If filter is shorter than necessary,
+	// it is expanded with transparent bytes
+	if (n >= m_mask.size()) 
+		MidiMsgFilter::setFilterLength(n-1);
+
+	m_template.at(n) |= (b << shl);
+}
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+void MidiMsgFilter::orMaskByte(unsigned int n, unsigned char b,
+							unsigned int shl){
+	
+	// If filter is shorter than necessary,
+	// it is expanded with transparent bytes
+	if (n >= m_mask.size()) 
+		MidiMsgFilter::setFilterLength(n-1);
+	
+	m_mask.at(n) |= (b << shl);
+}
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+void MidiMsgFilter::ignoreByte(unsigned int n){
+	if (n >= m_mask.size()) return;
+	m_mask.at(n) = 0;
+}
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+void MidiMsgFilter::setFilterLength(unsigned int fl){
+
+	unsigned int l = m_mask.size();
+	
+	// Truncate too long filters
+	for (; l > fl; l--){
+		m_template.pop_back();
+		m_mask.pop_back();
+	}
+
+	// Expand too short filters with transparent chunks (mask = 0)
+	for (; l < fl; l++){
+		m_template.push_back(0);
+		m_mask.push_back(0);
+	}
+}
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+void MidiMsgFilter::allowLongerMsg(){
+	m_allow_longer_msg = true;
+}
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+void MidiMsgFilter::disallowLongerMsg(){
+	m_allow_longer_msg = false;
+}
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+void MidiMsgFilter::setChannel(unsigned ch){
+
+	// If filter is completely empty, expand it to 1 byte
+	if (m_mask.size() < 1) setFilterLength(1);
+
+	if ((ch >=1) and (ch <=16)){
+		m_template[0] &= 0xF0;
+		m_template[0] += ch - 1;
+		m_mask[0]     |= 0x0F;
+	}
+	else {
+		m_mask[0]     &= 0xF0;
+	}
+}
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+bool MidiMsgFilter::check(MidiMsg& mm){
+
+	unsigned int l  = mm.getMessageLength();
+	unsigned int fl = m_mask.size();
+	unsigned char b;
+
+	// If message is shorter than a filter, it's clearly not a match
+	if (l < fl) return false;
+
+	// Message cannot be longer than a filter if m_allow_longer_msg isn't set
+	if ((l > fl) and !(m_allow_longer_msg)) return false;
```

Comment:
And that's my old Visual Basic habit from 1990s. ;) 
I'm not convinced whether replacing "and" with "&&" improves readability of code.  Probably makes no difference to most coders..

Reply:
> Put shortest to compute condition first.

Why is that? Won't all the conditions be evaluated anyway?

Reply:
> > Put shortest to compute condition first.
> 
> Why is that? Won't all the conditions be evaluated anyway?

Because you are using `and` which is a short-cut operator instead of `bitand`.
So if the first expression is false, it is useless to just even considering to evaluate the second one.

Otherwise to be coherent along https://en.cppreference.com/w/cpp/language/operator_alternative, what about
```suggestion
	if (not m_allow_longer_msg and l > fl) return false;
```
it looks more like Python and can run on computer from 1950 without `&` or `!` ;-)